stdlib/http/src/main/ballerina/src/http/caching/freshness_lifetime_calculation.bal

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/time;
+
+function isFreshResponse(Response cachedResponse, boolean isSharedCache) returns @tainted boolean {
+    int currentAge = getResponseAge(cachedResponse);
+    int freshnessLifetime = getFreshnessLifetime(cachedResponse, isSharedCache);
+    return freshnessLifetime > currentAge;
+}
+
+// Based on https://tools.ietf.org/html/rfc7234#section-4.2.1
+function getFreshnessLifetime(Response cachedResponse, boolean isSharedCache) returns int {
+    // TODO: Ensure that duplicate directives are not counted towards freshness lifetime.
+    var responseCacheControl = cachedResponse.cacheControl;
+    if (responseCacheControl is ResponseCacheControl) {
+        if (isSharedCache && responseCacheControl.sMaxAge >= 0) {
+            return responseCacheControl.sMaxAge;
+        }
+
+        if (responseCacheControl.maxAge >= 0) {
+            return responseCacheControl.maxAge;
+        }
+    }
+
+    // At this point, there should be exactly one Expires header to calculate the freshness lifetime.
+    // When adding heuristic calculations, the condition would change to >1.
+    if (cachedResponse.hasHeader(EXPIRES)) {
```

Comment:
Shall we do early return here by checking the negation to avoid nested if clauses?